### DISCUSSION TITLE
HarvestDirectory filtering of files.

### DISCUSSION THREAD

**Discussion by DSmitj** [wixtoolset/issues#7679] [Questions] [answered]
I was wondering if when using a harvestdirectory can files be filtered out in any other way other than by using a xsl file format?  I have some pdb files and config files that I don't want harvested.

I really like the integration with the VS IDE.  Thus why Iam asking the question.  I was hoping that an exclude could be used from within my project like in C#.

If not, is there an example xsl file format for WiX4 that I can reference or can a WiX3 xsl be transformed to WiX4?

Thanks,
D

**Comment by maniglia** [comment, marked as answer]
Hello @DSmitj, I can't answer your first question, but I can provide an xslt example.
This is the contents of the file .xslt, that removes all .pdb (be careful, it's not tested):

`<?xml version="1.0" encoding="utf-8"?>`
`<xsl:stylesheet`
`    xmlns:xsl="http://www.w3.org/1999/XSL/Transform"`
`    xmlns:wix="http://wixtoolset.org/schemas/v4/wxs"`
`    xmlns="http://wixtoolset.org/schemas/v4/wxs"`
`    version="1.0"`
`    exclude-result-prefixes="wix">`

`  <xsl:output method="xml" indent="yes" omit-xml-declaration="yes" />`

`  <xsl:strip-space elements="*" />`

`  <xsl:key`
`      name="PdbToRemove"`
`      match="wix:Component[ substring( wix:File/@Source, string-length( wix:File/@Source ) - 3 ) = '.pdb' ]"`
`      use="@Id"`
`    />`

`  <xsl:template match="@*|node()">`
`    <xsl:copy>`
`      <xsl:apply-templates select="@*|node()" />`
`    </xsl:copy>`
`  </xsl:template>`

`  <xsl:template match="*[ self::wix:Component or self::wix:ComponentRef ][ key( 'PdbToRemove', @Id ) ]" />`

`</xsl:stylesheet>`

I'm not a file .xslt expert.
Regards

**Reply by DSmitj** [reply]
Thank you

**Comment by nirbar** [comment]
You can use my WiX extension https://www.nuget.org/packages/PanelSwWixExtension4/
In your project file add:
```
<PropertyGroup>
     <HarvestDirectoryAdditionalOptions>-ext "$(PanelSwWixHarvestExtension)" -exc "*.pdb"</HarvestDirectoryAdditionalOptions>
<PropertyGroup>
```

**Reply by DSmitj** [reply]
Thank you
